File: backend/app/schemas/predict.py
from typing import Literal

from pydantic import BaseModel


class PeakProbabilityItem(BaseModel):
    peak_time: str
    prob_peak: float


class StationProbabilityResponse(BaseModel):
    station_id: str
    station_name: str
    base_time: str
    results: list[PeakProbabilityItem]
    max_prob: float
    max_prob_time: str


RiskLevel = Literal["low", "medium", "high", "critical", "unknown"]


class CurrentProbabilityStationItem(BaseModel):
    site_code: str
    station_name: str
    lat: float
    lon: float
    current_prob: float | None
    risk_level: RiskLevel


class CurrentProbabilitiesResponse(BaseModel):
    updated_at: str
    base_time: str
    is_stale: bool = False
    stations: list[CurrentProbabilityStationItem]

File: frontend/app/types/predict.ts
export interface PeakProbabilityItem {
  peak_time: string;
  prob_peak: number;
}

export interface StationProbabilityResponse {
  station_id: string;
  station_name: string;
  base_time: string;
  results: PeakProbabilityItem[];
  max_prob: number;
  max_prob_time: string;
}
